packages/common/src/lib/core/utils/utils.ts

```diff
@@ -11,7 +11,7 @@ export function getPackageLatestNpmVersion(pkg: string): string {
   }
 }
 
-export function getHttpProxyAgent(proxyUrl?: string): HttpProxyAgent | HttpsProxyAgent | undefined {
+export function getHttpProxyAgent(proxyUrl?: string, downloadUrl?: string): HttpProxyAgent | HttpsProxyAgent | undefined {
```

Comment:
can you please make `downloadUrl`  **not optional** (and so move it _before_  `proxyUrl` ?   as such:

```
export function getHttpProxyAgent(downloadUrl: string, proxyUrl?: string, ): HttpProxyAgent | HttpsProxyAgent | undefined {
```